process/peer/process.go

```diff
@@ -655,3 +684,51 @@ func (vs *validatorStatistics) getTempRating(s string) uint32 {
 
 	return peer.GetTempRating()
 }
+
+func (vs *validatorStatistics) decreaseAll(shardId uint32, missedRounds uint64, consensusGroupSize int) error {
+
+	log.Trace("ValidatorStatistics decreasing all", "shardId", shardId, "missedRounds", missedRounds)
+
+	shardValidators := vs.nodesCoordinator.GetAllValidatorsPublicKeys()[shardId]
+	validatorCount := len(shardValidators)
+	percentageRoundMissedFromTotalValidators := float64(missedRounds) / float64(validatorCount)
+	leaderAppearances := uint32(percentageRoundMissedFromTotalValidators + 1)
+	consensusGroupAppearances := uint32(float64(consensusGroupSize)*percentageRoundMissedFromTotalValidators + 1)
```

Comment:
I would replace this line with: 
consensusGroupAppearances := uint32(float64(consensusGroupSize)*percentageRoundMissedFromTotalValidators + 1 - math.SmallestNonzeroFloat64)
just to avoid the situation when float64(consensusGroupSize)*percentageRoundMissedFromTotalValidators would have a fixed value like (1, 2, ...). In this case is not necessary to increment the consensusGroupAppearances

Reply:
fixed